.github/workflows/unit_tests.yml

```diff
@@ -14,14 +14,14 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8  # v2
-      - name: Set up Python 3.8
+      - name: Set up Python 3.9
         uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435  # v4.5.0
         with:
-          python-version: 3.8
+          python-version: 3.9
       - name: Install Pip Dependencies
         run: pip install -r requirements.txt
       - name: Run Tests
-        run: nosetests
+        run: coverage run -m pytest
```

Comment:
Moved to pytest since we are moving to Python 3.9.  Our unit tests are really only using standard unit test stuff, so we shouldn't need to make any changes to the tests themselves.  Pytest is really just being used as the test discovery tool, so you execute it with `python -m pytest`, or `coverage run -m pytest`, and it should be fine.